databases/sync_tests/test_transactions.py

```diff
@@ -201,3 +201,68 @@ def test_transaction_already_closed(client: Prisma) -> None:
         transaction.user.delete_many()
 
     assert exc.match('Transaction already closed')
+
+
+@pytest.mark.parametrize(
+    ('input_level', 'expected_level'),
+    [
+        pytest.param(
+            prisma.TransactionIsolationLevel.READ_UNCOMMITTED,
+            'READ_UNCOMMITTED',
+            id='read uncommitted',
+            marks=pytest.mark.skipif(CURRENT_DATABASE in ['cockroachdb', 'sqlite'], reason='Not available'),
+        ),
+        pytest.param(
+            prisma.TransactionIsolationLevel.READ_COMMITTED,
+            'READ_COMMITTED',
+            id='read committed',
+            marks=pytest.mark.skipif(CURRENT_DATABASE in ['cockroachdb', 'sqlite'], reason='Not available'),
+        ),
+        pytest.param(
+            prisma.TransactionIsolationLevel.REPEATABLE_READ,
+            'REPEATABLE_READ',
+            id='repeatable read',
+            marks=pytest.mark.skipif(CURRENT_DATABASE in ['cockroachdb', 'sqlite'], reason='Not available'),
+        ),
+        pytest.param(
+            prisma.TransactionIsolationLevel.SNAPSHOT,
+            'SNAPSHOT',
+            id='snapshot',
+            marks=pytest.mark.skipif(True, reason='Available for SQL Server only'),
+        ),
+        pytest.param(
+            prisma.TransactionIsolationLevel.SERIALIZABLE,
+            'SERIALIZABLE',
+            id='serializable',
+            marks=pytest.mark.skipif(
+                CURRENT_DATABASE == 'sqlite', reason='PRAGMA has only effect in shared-cache mode'
+            ),
+        ),
+    ],
+)
+@pytest.mark.skipif(
+    CURRENT_DATABASE in ['mysql', 'mariadb'],
+    reason="""
+    MySQL 8.0 doesn't have the way to query the current transaction isolation level.
+    See https://bugs.mysql.com/bug.php?id=53341
+
+    Refs:
+    * https://github.com/prisma/prisma/issues/22890
+    """,
+)
+def test_isolation_level(
+    client: Prisma, raw_queries: RawQueries, input_level: prisma.TransactionIsolationLevel, expected_level: str
+) -> None:
+    """Ensure that transaction isolation level is set correctly"""
+    with client.tx(isolation_level=input_level) as tx:
+        results = tx.query_raw(raw_queries.select_tx_isolation)
+
+    assert len(results) == 1
+
+    row = results[0]
+    assert any(row)
+
+    level = next(iter(row.values()))
+    # The result can depends on the database, so we do upper() and replace()
+    level = str(level).upper().replace(' ', '_').replace('-', '_')
+    assert level == expected_level
```

Comment:
minor nit (can leave as is): I think it would make sense to instead of defining `expected_level` as a single string, it is a dictionary that contains that's a map of database -> expected, e.g.

```py
pytest.param(
    prisma.TransactionIsolationLevel.SERIALIZABLE,
    {'postgresql': 'SERIALIZABLE', **etc},
    id='serializable',
    marks=pytest.mark.skipif(
        CURRENT_DATABASE == 'sqlite', reason='PRAGMA has only effect in shared-cache mode'
    ),
),
# ...
assert level == expected_levels[CURRENT_DATABASE]
```